htlcswitch/link.go

```diff
@@ -3554,6 +3567,49 @@ func (l *channelLink) processRemoteAdds(fwdPkg *channeldb.FwdPkg,
 	l.forwardBatch(replay, switchPackets...)
 }
 
+// experimentalEndorsement returns the value to set for our outgoing
+// experimental endorsement field, and a boolean indicating whether it should
+// be populated on the outgoing htlc.
+func (l *channelLink) experimentalEndorsement(
+	customUpdateAdd record.CustomSet) ([]byte, bool) {
+
+	// Only relay experimental signal if we are within the experiment
+	// period.
+	if !l.cfg.ForwardExperimentalEndorsement() {
+		return nil, false
+	}
+
+	// If we don't have any custom records or the experimental field is
+	// not set, just forward a zero value.
+	if len(customUpdateAdd) == 0 {
+		return []byte{0}, true
+	}
+
+	value, set := customUpdateAdd[lnwire.ExperimentalEndorsementType]
+	if !set {
+		return []byte{0}, true
+	}
+
+	// We expect at least one byte for this field, consider it invalid if
+	// it has no data and just forward a zero value.
+	if len(value) == 0 {
```

Comment:
I think it can be commented that more than one byte is considered as a valid input to give more flexibility (see Postel's law "be conservative in you do, be liberal in what you accept from others").

I don't know if `ExperimentalEndorsementType` is well-typed L22 in `lnwire/update_add_htlc.go` (i.e `byte`, `uint32_t`, `int64` ?).

Reply:
This only invalidates the 0 byte case which I think makes sense. This seems like a sensible interpretation.